Clarify `button_to` helper changes in Rails 7.0 (#43855)
rails/rails: guides/source/upgrading_ruby_on_rails.md

Upgrading from Rails 6.1 to Rails 7.0
-------------------------------------

### `ActionView::Helpers::UrlHelper#button_to` changed behavior

> If your application uses `button_to` helper with an Array that contains an Active Record object as the last element. Starting from Rails 7.0 `button_to` renders a `form` tag with `patch` HTTP verb for a persisted object.

Reviewer: Hmmm, it looks like this happens even if you don't pass an array, according to the example in https://github.com/rails/rails/pull/43413

Reviewer: ```suggestion
Starting from Rails 7.0 `button_to` renders a `form` tag with `patch` HTTP verb if any of its arguments are a persisted ActiveRecord object.
```

Author: Changed
